File: backend/src/main/java/edu/cit/cabasag/barberconnect/feature/social/Post.java
package edu.cit.cabasag.barberconnect.feature.social;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

/**
 * Vertical Slice Architecture — Social Feature
 * Domain model for barber posts in the feed.
 */
@Data @NoArgsConstructor @AllArgsConstructor
public class Post {
    private String post_id;
    private String barber_profile_id;
    private String barberUserId;
    private String barberFullName;
    private String barberProfileImageUrl;
    private String barberShopName;
    private String content;
    private String imageUrl;
    private Integer likesCount = 0;
    private Integer commentsCount = 0;
    private Boolean isActive = true;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private List<String> commentIds;
    private List<String> reactionIds;
}

File: mobile/app/src/main/java/edu/cit/cabasag/barberconnect/model/Post.kt
package edu.cit.cabasag.barberconnect.model

import com.google.gson.annotations.SerializedName

data class Post(
    @SerializedName("post_id") val postId: String? = null,
    @SerializedName("barber_profile_id") val barberProfileId: String? = null,
    @SerializedName(value = "user_id", alternate = ["barberUserId", "barber_user_id"]) val userId: String? = null,
    @SerializedName(value = "barberFullName", alternate = ["barber_full_name", "barberName", "barber_name"]) val barberFullName: String? = null,
    @SerializedName(value = "barberProfileImageUrl", alternate = ["barber_profile_image_url"]) val barberProfileImageUrl: String? = null,
    @SerializedName(value = "barberShopName", alternate = ["barber_shop_name", "shopName"]) val barberShopName: String? = null,
    val content: String? = null,
    @SerializedName(value = "imageUrl", alternate = ["image_url"]) val imageUrl: String? = null,
    @SerializedName(value = "likesCount", alternate = ["likes_count"]) val likesCount: Int? = null,
    @SerializedName(value = "commentsCount", alternate = ["comments_count"]) val commentsCount: Int? = null,
    @SerializedName(value = "isActive", alternate = ["is_active"]) val isActive: Boolean? = null,
    @SerializedName(value = "createdAt", alternate = ["created_at"]) val createdAt: String? = null,
    @SerializedName(value = "updatedAt", alternate = ["updated_at"]) val updatedAt: String? = null
)
